[Varun Shukla] iP

Shux347 opened the pull request:
**Sam (Super Amazing Machine)**


>Sam is designed to enhance your communication experience by engaging in meaningful and insightful conversations. 

It’s:

•	text-based
•	intuitive to use
•	FAST and RELIABLE

All you need to do is:

1.	Clone the repository from [here](https://github.com/nus-cs2103-AY2425S1/ip/pull/319#). :point_left:
2.	Run the application with your preferred setup.
3.	Interact with Sam to explore its features.

And guess what? It’s FREE!

Features:

- [x] 	 Managing tasks
- [x]  Managing deadlines
- [x]  Managing events
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here’s a sample of the `main` method that kicks off Sam:
```java
public static void main(String[] args) {
        new Sam("data/Sam.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Shux347 committed:
Finished level-0 increment

Shux347 committed:
Finished level-1 increment

Shux347 committed:
Finished level-2 increment

Shux347 committed:
Finished level-3 increment

Shux347 committed:
Finished level-4 increment

Shux347 committed:
Finished A-TextUiTesting increment

Shux347 committed:
Finished level-5 increment

Shux347 committed:
Finished level-6 increment

Shux347 committed:
Include DataHandler class and Sam.txt data file

Shux347 committed:
Add support for date object for deadlines

Shux347 committed:
Add documentation for DataHandler and Items classes

Shux347 committed:
Include new classes

Shux347 committed:
Add into package sam

Shux347 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Shux347 committed:
Add one test case for Item class and Sam class

Shux347 committed:
Update gradle and gitignore

Shux347 committed:
Change testcase name to follow coding standards

Shux347 committed:
Add find function

Shux347 committed:
Add javadoc for test case

Shux347 committed:
Merge branch 'branch-A-CodingStandard'

Shux347 committed:
Merge branch 'branch-Level-9'

Shux347 committed:
Add checkstyle

Shux347 committed:
Add javafx

arnav12344 commented on src/main/java/sam/Sam.java:
```diff
@@ -0,0 +1,228 @@
+     * @param filePath The file path to the storage file.
+     */
+    public Sam() {
+        String filePath = "data/Sam.txt";
```
Should this be a constant?

RobotWizzard replied:
I disagree with @arnav12344 's comment here. Since it is only used in once in the entire program, I think it can just be hardcoded:
```
storage = new Storage("data/Sam.txt");
```

arnav12344 commented on src/main/java/sam/Event.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    @Override
+    public String toData() {
```
Should this method name be more specific?

arnav12344 reviewed (commented):
I like how clean and well written your code is!

RobotWizzard commented on src/main/java/sam/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+     * @param newname The name of the deadline task.
+     * @param by The deadline of the task in the format "dd-MM-yyyy".
+     */
+    public Deadline(String newname, String by) {
```
I didn't like the name "newname" for the parameter here, because:
1. Variable names should be in camelCase (e.g. "newName")
2. I think just "name" is sufficient and self-explanatory for the parameter. Parameters for subclasses don't have to differ from their parent classes.

I noticed the same issue in other classes too.

RobotWizzard commented on src/main/java/sam/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+    public Deadline(String newname, String by) {
+        super(newname);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+        this.by = LocalDate.parse(by, formatter);
```
I don't think the variable "formatter" here is necessary, as it is only used in the next line it is declared. Perhaps 
```
this.by = LocalData.parse(by, DateTimeFormatter.ofPattern("dd-MM-yyyy"));
```
is already quite readable.

RobotWizzard commented on src/main/java/sam/Event.java:
```diff
@@ -0,0 +1,40 @@
+ */
+public class Event extends Item {
+
+    private static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("dd-MM-yyyy");
```
I like how you stored the date format as a constant here! Maybe you can do the same in the "Deadline" class to make it more consistent.

RobotWizzard commented on src/main/java/sam/Item.java:
```diff
@@ -0,0 +1,65 @@
+    /**
+     * Marks the item as undone.
+     */
+    public void markAsUndone() {
```
I didn't like the name of this method since it is too similar to "markAsDone" to convey that this does the opposite. Furthermore, "undone" is not actually an English word. Perhaps "markAsNotDone" or "unmark" would be better names here.

RobotWizzard commented on src/main/java/sam/Items.java:
```diff
@@ -0,0 +1,113 @@
+/**
+ * Represents a task list.
+ */
+public class Items {
```
Not sure if I like the name "Items" for this class, as it is very similar to the "Item" class. Since the header comment states that this represents a task list, maybe "TasksList" is more appropriate?

RobotWizzard commented on src/main/java/sam/Sam.java:
```diff
@@ -0,0 +1,228 @@
+
+    /**
+     * Handles a single user input and returns the response as a string.
+     * @param input the input from the user
```
Should there be an empty line before this?

RobotWizzard commented on src/main/java/sam/Sam.java:
```diff
@@ -0,0 +1,228 @@
+                if (items.getSize() > 0) {
+                    storage.save(items.getItems());
+                }
+
```
I think this is just an accidental empty line. 😳

RobotWizzard commented on src/main/java/sam/SamException.java:
```diff
@@ -0,0 +1,24 @@
+
+    /**
+     * Constructs a new SamException with a default detail message.
+     * The default message is "Huh? I'm not quite sure what you mean by that."
```
I like that you included the default message in the header comment. Nice! 👍

RobotWizzard commented on src/main/java/sam/Storage.java:
```diff
@@ -0,0 +1,96 @@
+            String name = data[2];
+
+            switch (itemType) {
+            case "T":
+                ToDo todo = new ToDo(name);
+                if (isDone) {
+                    todo.markAsDone();
+                }
+                items.add(todo);
+                break;
+            case "D":
+                Deadline deadline = new Deadline(name, data[3]);
+                if (isDone) {
+                    deadline.markAsDone();
+                }
+                items.add(deadline);
+                break;
+            case "E":
+                Event event = new Event(name, data[3], data[4]);
+                if (isDone) {
+                    event.markAsDone();
+                }
+                items.add(event);
+                break;
+            default:
+                throw new IOException("Unknown item type in storage file.");
+            }
```
Should this part be indented once more? It is in the switch block.

RobotWizzard reviewed (commented):
I thought that your code was really neat and organized 👍, there were just some tiny bits about naming to fix. 😄

Shux347 committed:
Add assert statements

Assert that input from scanner is not null or empty as this is not an expected response

Shux347 committed:
Improve code quality

Add private enum class CommandType to handle inputs better
Use switch cases to determine command type

Shux347 committed:
Merge pull request #1 from Shux347/branch-A-Assertations

Add assert statements

Shux347 committed:
Merge branch 'master' into branch-A-CodeQuality

Shux347 committed:
Merge pull request #2 from Shux347/branch-A-CodeQuality

Improve code quality

Shux347 committed:
Add help command

Type help to see list of commands so that user knows how to use the bot

Shux347 committed:
Improve code quality

Take into account comments from PR
Fix issues flagged by auto checker

Shux347 committed:
Add Ui.png

Add chatbot name in window bar
Upload Ui.png

Shux347 committed:
Add User Guide
